test/components/popup.spec.tsx

```diff
@@ -28,68 +50,82 @@ describe('<Popup />', function() {
 
     describe('The popup user', function() {
         it('clicks on the parent and the popup opens and closes after another click', async function() {
-            const {select, waitForDom} = clientRenderer.render(<PopupDemo />);
+            const {result, waitForDom} = clientRenderer.render(<PopupDemo />);
+
+            let popupDemo = new PopupDemoTestDriver(result as PopupDemo);
 
             await waitForDom(() => {
-                expect(select(demoContainer)).to.be.present();
-                expect(select(demoContainer, portalId)).to.be.absent();
+                expect(popupDemo.container).to.be.present();
+                expect(popupDemo.isPopupPresent).to.be.false;
             });
-            select<HTMLDivElement>(demoContainer)!.click();
-            await waitForDom(() => expect(bodySelect(portalId)).to.be.present());
-            select<HTMLDivElement>(demoContainer)!.click();
-            return waitForDom(() => expect(bodySelect(portalId)).to.be.absent());
+
+            (popupDemo.container as HTMLDivElement).click();
+            popupDemo = new PopupDemoTestDriver(result as PopupDemo);
+            await waitForDom(() => expect(popupDemo.portal).to.be.present());
+
+            (popupDemo.container as HTMLDivElement).click();
+            popupDemo = new PopupDemoTestDriver(result as PopupDemo);
+            return waitForDom(() => expect(popupDemo.isPopupPresent).to.be.false);
         });
     });
 
     it('displays the popup and renders its children if the open prop is given', function() {
-        clientRenderer.render(
+        const {result} = clientRenderer.render(
             <Popup anchor={anchor} open>
-                <span data-automation-id="SPAN">Popup Body</span>
+                <span>Popup Body</span>
             </Popup>
         );
 
+        const popup = new PopupTestDriver(result as Popup);
+
         return waitFor(() => {
-            expect(bodySelect(portalId)).to.be.present();
-            expect(bodySelect(portalId, 'SPAN')).to.be.present();
+            expect(popup.root).to.be.present();
```

Comment:
why not use `isPresent()`. this is relevant to all places where you go  `expect(popup.root).to.be.present()`

Reply:
`to.be.present()` checks for many things when it comes to the question of being present on the screen.
Perhaps I should rename `isPresent()` to distinguish between them